collect_app/src/main/java/org/odk/collect/android/application/initialization/MetaPreferenceMigrator.java

```diff
@@ -32,26 +31,22 @@
 /**
  * Migrates old preference keys and values to new ones.
  */
-public class CollectPreferenceMigrator implements PreferenceMigrator {
+public class MetaPreferenceMigrator implements PreferenceMigrator {
```

Comment:
Not for testing it's just me playing with the interface. I've flipped-flopped on this a few times. I guess my thinking here is that in the `SettingsImporter` we're actively dealing with general and admin preferences so it's weird that we then just call `migrate`. There w know what we're trying to migrate we just don't care about the detail (and we also don't know that the meta prefs exist). 

I think it's also me not being able to escape from the discussions about having profiles in the future. In that world you'd have different general and shared prefs but one shared meta prefs. 

Any of that make sense or you still hate it?

Reply:
The rationale for passing in general and admin preferences to migrate makes sense to me now, thanks. The name still really bothers me. `CollectPreferenceMigrator` or `SharedPrefsMigrator` or something that implies it can migrate all the prefs would make a lot more sense to me. Even if it doesn't keep track of general and admin preferences, it still knows how to migrate all the relevant preferences.

Reply:
Yeah that's very fair. I'll rework the name.